src/Symfony/Component/Process/Pipes/WindowsPipes.php

```diff
@@ -47,14 +47,24 @@ public function __construct($disableOutput, $input)
             // Workaround for this problem is to use temporary files instead of pipes on Windows platform.
             //
             // @see https://bugs.php.net/bug.php?id=51800
-            $this->files = array(
-                Process::STDOUT => tempnam(sys_get_temp_dir(), 'out_sf_proc'),
-                Process::STDERR => tempnam(sys_get_temp_dir(), 'err_sf_proc'),
+            $pipes = array(
+                Process::STDOUT => Process::OUT,
+                Process::STDERR => Process::ERR,
             );
-            foreach ($this->files as $offset => $file) {
-                if (false === $file || false === $this->fileHandles[$offset] = @fopen($file, 'rb')) {
-                    throw new RuntimeException('A temporary file could not be opened to write the process output to, verify that your TEMP environment variable is writable');
+            $tmpDir = sys_get_temp_dir();
+            if (!@fopen($tmpDir.'\\sf_proc_00.check', 'wb')) {
+                throw new RuntimeException('A temporary file could not be opened to write the process output to, verify that your TEMP environment variable is writable');
+            }
+            for ($i = 0;; ++$i) {
+                foreach ($pipes as $pipe => $name) {
+                    @unlink($file = sprintf('%s\\sf_proc_%02X.%s', $tmpDir, $i, $name));
```

Comment:
shouldn't you check whether ``$this->files[$pipe]`` is already set, to avoid reopening the same pipe several times in case of a retry when the other one fails ? Or is it done on purpose ?